Globals.cpp

```diff
@@ -746,12 +746,55 @@ global_settings::ConfigParse(cParser &Parser) {
         }
         else if (token == "uartfeature")
         {
-            Parser.getTokens(4);
-            Parser
-                >> uart_conf.mainenable
-                >> uart_conf.scndenable
-                >> uart_conf.trainenable
-                >> uart_conf.localenable;
+            Parser.getTokens(1);
+            std::string firstToken;
+            Parser >> firstToken;
+
+            std::map<std::string, bool*> features_map = {
+                {"main", &uart_conf.mainenable},
+                {"scnd", &uart_conf.scndenable},
+                {"train", &uart_conf.trainenable},
+                {"local", &uart_conf.localenable},
+                {"radiovolume", &uart_conf.radiovolumeenable},
+                {"radiochannel", &uart_conf.radiochannelenable}
+            };
+
+            if(firstToken.find('|') != std::string::npos || features_map.count(firstToken)) {
+                // new format (features delimited by pipe)
+
+                // all settings should be disabled initially
+                for(auto const &x : features_map) {
+                    *(x.second) = false;
+                }
+
+                // and only defined should be enabled
+                std::string key;
+                std::stringstream firstTokenStream = std::stringstream(firstToken);
+
+                while(!firstTokenStream.eof()) {
+                    std::getline(firstTokenStream, key, '|');
+
+                    if(features_map.count(key)) {
+                        *(features_map[key]) = true;
+                    }
+                }
+            } else {
+                // legacy format
+
+                // first token must be casted to bool
+                uart_conf.mainenable = (
```

Comment:
Prefer not to duplicate parser logic, replace `Parser >> firstToken` with `firstToken = Parser.peek()` and do `Parser >> uart_conf.mainenable` here.

Reply:
I prefer DRY. I haven't noticed peek() before. Thanks. 👍